gateway-service/src/test/java/org/java/mentorship/gateway/controller/SessionControllerTest.java

```diff
@@ -0,0 +1,76 @@
+package org.java.mentorship.gateway.controller;
+
+import org.java.mentorship.contracts.user.client.UserFeignClient;
+import org.java.mentorship.contracts.user.dto.Session;
+import org.java.mentorship.contracts.user.dto.request.LoginRequest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+import java.util.List;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class SessionControllerTest {
+
+    @Mock
+    private UserFeignClient userFeignClient;
+
+    @InjectMocks
+    private SessionController sessionController;
+
+    @Test
+    void createSessionShouldReturnDataFromFeign() {
+        when(userFeignClient.createSession(any(LoginRequest.class)))
+                .thenReturn(Session.builder().build());
+
+        ResponseEntity<Session> response = sessionController.createSession(new LoginRequest());
+
+        assertEquals(200, response.getStatusCode().value());
+        assertTrue(Objects.nonNull(response.getBody()));
+    }
+
+    @Test
+    void getSessionShouldReturnDataFromFeign() {
+        when(userFeignClient.getSession(anyString()))
+                .thenReturn(Session.builder().build());
+
+        ResponseEntity<Session> response = sessionController.getSession(anyString());
+
+        assertEquals(200, response.getStatusCode().value());
+        assertTrue(Objects.nonNull(response.getBody()));
+    }
+
+    @Test
+    void getSessionsByUserShouldReturnDataFromFeign() {
+        when(userFeignClient.getSessions(anyInt()))
+                .thenReturn(List.of(Session.builder().build()));
```

Comment:
Using List.of for just one element has performance issues

Please use Collections.singletonList(element)